### PR TITLE
lib/streamaggr: reduce number of inuse objects

### PR DESCRIPTION
**Pull request by hagen1778**
The main change is getting rid of interning of sample key. It was discovered that for cases with many unique time series aggregated by vmagent interned keys could grow up to hundreds of millions of objects. This has negative impact on the following aspects:
1. It slows down garbage collection cycles, as GC has to scan all inuse objects periodically. The higher is the number of inuse objects, the longer it takes/the more CPU it takes.
2. It slows down the hot path of samples aggregation where each key needs to be looked up in the map first.

The change makes code more fragile, but suppose to provide performance optimization for heavy-loaded vmagents with stream aggregation enabled.

### CODE REVIEW

File: lib/streamaggr/streamaggr.go [hunk 1]
```diff
@@ -802,7 +802,7 @@ func (a *aggregator) Push(tss []prompbmarshal.TimeSeries, matchIdxs []byte) {
 	defer putPushCtx(ctx)
 
 	samples := ctx.samples
-	buf := ctx.buf
+	buf := ctx.buf[:0]
```
Comment (AndrewChubatiuk) [inline lib/streamaggr/streamaggr.go:805]: it's already done here 
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/bff83230631437988939d5891e3f8a94e9ba9c40/lib/streamaggr/streamaggr.go#L928

**Comment by hagen1778** [comment]
See test conducted against v1.101.0 vmagent (`vmagent-aggregator-old`) and vmagent with patch from this PR (`vmagent-aggregator`). Both vmagents were receiving about ~55k samples/s aggregating them all. The observations are following:
image

The CPU usage remains the same, but new patched vmagent looks more stable. The difference between number of allocate objects in heap is x2. This difference will have bigger impact for higher workload, reducing time spent on GC scans.

**Comment by hagen1778** [comment]
Tests after https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6402/commits/d80c1c88888970993c75fa46eff96c363f0f4853 at higher load (~250k samples/s):

image
image

**Review by valyala — APPROVED** [review]

**Comment by valyala** [comment]
@hagen1778 , thanks for the pull request!

**Comment by valyala** [comment]
Please see the follow-up commit 36be090cd521f9f80558b1284a9f093c6e85240f

**Comment by valyala** [comment]
FYI, this pull request has been included in [v1.102.0-rc1 release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.102.0-rc1).

**Pull request merged**